### PR TITLE
Add Varys Keyboard

### PR DESCRIPTION
**Pull request by vinaykomaravolu**
## Description

Add Varys keyboard to QMK repo

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/varys/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+ * |  17  |  16  |  15  |  14  |  13  |  12  |-------.    ,-------|  40  |  41  |  42  |  43  |  44  |  45  |
+ * |------+------+------+------+------+------| NO LED|    |NO LED |------+------+------+------+------+------|
+ * |  23  |  22  |  21  |  20  |  19  |  18  |-------|    |-------|  46  |  47  |  48  |  49  |  50  |  51  |
+ * `-----------------------------------------/       /     \      \-----------------------------------------'
+ *                   | 27   |26    |25   |  /24     /       \52    \  |53    |54     |55    |
+ *                   |      |      |     | /       /         \      \ |      |      |      |
+ *                   `----------------------------'           '------''--------------------'         
+ *  {64,0}                                                                                           {224,64}      
+ */
+const is31fl3737_led_t PROGMEM g_is31fl3737_leds[IS31FL3737_LED_COUNT] = {
```
Comment (drashna) [inline keyboards/varys/keymaps/default/keymap.c:79]: This should not be in a keymap.  This should be in `varys.c`, alongside keyboard.json.

File: keyboards/varys/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+    {0, SW9_CS4, SW7_CS4, SW8_CS4 },
+    {0, SW9_CS5, SW7_CS5, SW8_CS5 },
+    {0, SW9_CS6, SW7_CS6, SW8_CS6 },
+    {0, SW9_CS7, SW7_CS7, SW8_CS7 },
+    {0, SW9_CS8, SW7_CS8, SW8_CS8 },
+    {0, SW9_CS9, SW7_CS9, SW8_CS9 },
+    {0, SW9_CS10, SW7_CS10, SW8_CS10 }, // 28 LED right
+}; // 56 LED total
+
+led_config_t g_led_config = {
```
Comment (drashna) [inline keyboards/varys/keymaps/default/keymap.c:146]: This should be converted to json.

File: keyboards/varys/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+    led_t led_state = host_keyboard_led_state();
+    oled_set_cursor(0, 9);
+    oled_write_P(led_state.num_lock ? PSTR(" NUM ") : PSTR("     "), false);
+    oled_set_cursor(0, 10);
+    oled_write_P(led_state.caps_lock ? PSTR(" CAP ") : PSTR("     "), false);
+    oled_set_cursor(0, 11);
+    oled_write_P(led_state.scroll_lock ? PSTR(" SCR ") : PSTR("     "), false);
+    return false;
+}
+#endif
```
Comment (drashna) [inline keyboards/varys/keymaps/default/keymap.c:254]: This should also be moved to varys.c, and use `oled_task_kb` instead.

File: keyboards/varys/rules.mk [hunk 4]
```diff
@@ -0,0 +1,5 @@
+ENCODER_ENABLE = yes
+OLED_ENABLE = yes
+OLED_DRIVER = ssd1306
+OLED_TRANSPORT = i2c
```
Comment (drashna) [inline keyboards/varys/rules.mk:4]: These are the default, and don't need to be set, despite what stuff like chatgpt says. 
```suggestion
```

File: keyboards/varys/rules.mk
Comment (drashna) [inline keyboards/varys/rules.mk:1]: All of these should be moved to json.

File: keyboards/varys/config.h
Comment (drashna) [inline keyboards/varys/config.h:1]: All of this, but the "low is left" can and should be moved to info.json.  

And commented out code should either be included or should be removed.

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by vinaykomaravolu** [comment]
Thank you for the comments. Will make the changes and resubmit.

**Comment by vinaykomaravolu** [comment]
Updated PR with drashna's comments

File: keyboards/varys/keyboard.json [hunk 5]
```diff
@@ -0,0 +1,118 @@
+{
+    "manufacturer": "Vinay Komaravolu",
+    "keyboard_name": "Varys",
+    "maintainer": "Vinay Komaravolu",
+    "bootloader": "caterina",
```
Comment (waffle87) [inline keyboards/varys/keyboard.json:5]:
```suggestion
    "development_board": "promicro",
```

File: keyboards/varys/keyboard.json [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+        "mousekey": true,
+        "nkro": true,
+        "oled": true,
+        "encoder": true
+    },
+    "matrix_pins": {
+        "rows": ["B6", "B7", "C6", "C7", "D4"],
+        "cols": ["B5", "B4", "B3", "B2", "B1", "B0"]
+    },
+    "processor": "atmega32u4",
```
Comment (waffle87) [inline keyboards/varys/keyboard.json:22]:
```suggestion
```

File: keyboards/varys/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+                               _______, _______, _______, _______, _______,  _______, _______, _______
+    ),
+    [_MISC] = LAYOUT(
+    _______, _______, _______, _______, _______, _______,                   _______, _______, _______, _______, _______, _______,
+    _______, _______, _______, _______, _______, _______,                   _______, _______, _______, _______, _______, _______,
+    _______, _______, _______, _______, _______, _______,                   _______, _______, _______, _______, _______, _______,
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+                               _______, _______, _______, _______, _______,  _______, _______, _______
+    )
```
Comment (waffle87) [inline keyboards/varys/keymaps/default/keymap.c:58]:
```suggestion
    [_RAISE] = LAYOUT(
    QK_BOOT, _______, _______, _______, _______, _______,                   _______, _______, _______, _______, _______, _______,
    _______, _______, _______, _______, _______, _______,                   _______, _______, _______, _______, _______, _______,
    _______, _______, _______, _______, _______, _______,                   _______, _______, _______, _______, _______, _______,
    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
                               _______, _______, _______, _______, _______,  _______, _______, _______
    )
```

File: keyboards/varys/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,76 @@
+// Copyright 2023 Vinay Komaravolu
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+
+enum layer_number {
+    _QWERTY = 0,
+    _RAISE,
+    _LOWER,
+    _MISC,
```
Comment (waffle87) [inline keyboards/varys/keymaps/default/keymap.c:10]:
```suggestion
    _QWERTY,
    _RAISE,
    _LOWER
```

File: keyboards/varys/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+                               _______, _______, _______, _______, _______,  _______, _______, _______
+    )
+};
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+    state = update_tri_layer_state(state, _RAISE, _LOWER, _MISC);
+    return state;
+}
+
```
Comment (waffle87) [inline keyboards/varys/keymaps/default/keymap.c:65]: The [tri-layer](https://docs.qmk.fm/features/tri_layer) feature should be used instead.
```suggestion
```

File: keyboards/varys/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+enum layer_number {
+    _QWERTY = 0,
+    _RAISE,
+    _LOWER,
+    _MISC,
+};
+
+#define L_RAISE MO(_RAISE)
+#define L_LOWER MO(_LOWER)
+
```
Comment (waffle87) [inline keyboards/varys/keymaps/default/keymap.c:15]: As per above, the `TL_LOWR` & `TL_UPPR` keycodes should be used instead.
```suggestion
```

File: keyboards/varys/keymaps/default/keymap.c [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+ *                   | LGui | LAlt |Enter | /Space  /       \Enter \  |BackSP |RAlt | RGUI |
+ *                   |      |      |      |/       /         \      \ |      |      |      |
+ *                   `----------------------------'           '------''--------------------'
+ */
+    [_QWERTY] = LAYOUT(
+    KC_ESC,   KC_1,   KC_2,    KC_3,    KC_4,    KC_5,                              KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_GRV,
+    KC_TAB,   KC_Q,   KC_W,    KC_E,    KC_R,    KC_T,                              KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_MINS,
+    KC_LSFT,  KC_A,   KC_S,    KC_D,    KC_F,    KC_G,                              KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,
+    KC_LCTL,  KC_Z,   KC_X,    KC_C,    KC_V,    KC_B, KC_WHOM,    KC_AUDIO_MUTE,   KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,  KC_RSFT,
+                               KC_LALT, L_LOWER,KC_LGUI, KC_SPC,   KC_ENT, KC_BSPC,  L_RAISE, KC_RALT
```
Comment (waffle87) [inline keyboards/varys/keymaps/default/keymap.c:36]:
```suggestion
                               KC_LALT, TL_LOWR,KC_LGUI, KC_SPC,   KC_ENT, KC_BSPC,  TL_UPPR, KC_RALT
```

File: keyboards/varys/keymaps/default/keymap.c [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+#ifdef ENCODER_MAP_ENABLE
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
+    [_QWERTY] = {ENCODER_CCW_CW(KC_BRIGHTNESS_DOWN, KC_BRIGHTNESS_UP),
+                 ENCODER_CCW_CW(KC_VOLD, KC_VOLU)},
+    [_RAISE] = {ENCODER_CCW_CW(_______, _______),
+                ENCODER_CCW_CW(_______, _______)},
+    [_LOWER] = {ENCODER_CCW_CW(_______, _______),
+                ENCODER_CCW_CW(_______, _______)},
+    [_MISC] = {ENCODER_CCW_CW(_______, _______),
+               ENCODER_CCW_CW(_______, _______)}};
```
Comment (waffle87) [inline keyboards/varys/keymaps/default/keymap.c:75]:
```suggestion
    [_LOWER] = {ENCODER_CCW_CW(_______, _______),
                ENCODER_CCW_CW(_______, _______)}
};
```

File: keyboards/varys/keymaps/via/keymap.c [hunk 13]
```diff
@@ -0,0 +1,76 @@
+// Copyright 2023 Vinay Komaravolu
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include QMK_KEYBOARD_H
+
+enum layer_number {
+    _QWERTY = 0,
+    _RAISE,
+    _LOWER,
+    _MISC,
```
Comment (waffle87) [inline keyboards/varys/keymaps/via/keymap.c:10]:
```suggestion
    _QWERTY,
    _RAISE,
    _LOWER
```

File: keyboards/varys/keymaps/via/keymap.c [hunk 14]
```diff
@@ -0,0 +1,76 @@
...
+};
+
+#define L_RAISE MO(_RAISE)
+#define L_LOWER MO(_LOWER)
+
```
Comment (waffle87) [inline keyboards/varys/keymaps/via/keymap.c:15]:
```suggestion
```

File: keyboards/varys/keymaps/via/keymap.c [hunk 15]
```diff
@@ -0,0 +1,76 @@
...
+ *                   | LGui | LAlt |Enter | /Space  /       \Enter \  |BackSP |RAlt | RGUI |
+ *                   |      |      |      |/       /         \      \ |      |      |      |
+ *                   `----------------------------'           '------''--------------------'
+ */
+    [_QWERTY] = LAYOUT(
+    KC_ESC,   KC_1,   KC_2,    KC_3,    KC_4,    KC_5,                              KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_GRV,
+    KC_TAB,   KC_Q,   KC_W,    KC_E,    KC_R,    KC_T,                              KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_MINS,
+    KC_LSFT,  KC_A,   KC_S,    KC_D,    KC_F,    KC_G,                              KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,
+    KC_LCTL,  KC_Z,   KC_X,    KC_C,    KC_V,    KC_B, KC_WHOM,    KC_AUDIO_MUTE,   KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,  KC_RSFT,
+                               KC_LALT, L_LOWER,KC_LGUI, KC_SPC,   KC_ENT, KC_BSPC,  L_RAISE, KC_RALT
```
Comment (waffle87) [inline keyboards/varys/keymaps/via/keymap.c:36]:
```suggestion
                               KC_LALT, TL_LOWR,KC_LGUI, KC_SPC,   KC_ENT, KC_BSPC,  TL_UPPR, KC_RALT
```

File: keyboards/varys/keymaps/via/keymap.c [hunk 16]
```diff
@@ -0,0 +1,76 @@
...
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+                               _______, _______, _______, _______, _______,  _______, _______, _______
+    ),
+    [_MISC] = LAYOUT(
+    _______, _______, _______, _______, _______, _______,                   _______, _______, _______, _______, _______, _______,
+    _______, _______, _______, _______, _______, _______,                   _______, _______, _______, _______, _______, _______,
+    _______, _______, _______, _______, _______, _______,                   _______, _______, _______, _______, _______, _______,
+    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+                               _______, _______, _______, _______, _______,  _______, _______, _______
+    )
```
Comment (waffle87) [inline keyboards/varys/keymaps/via/keymap.c:58]:
```suggestion
    [_RAISE] = LAYOUT(
    QK_BOOT, _______, _______, _______, _______, _______,                   _______, _______, _______, _______, _______, _______,
    _______, _______, _______, _______, _______, _______,                   _______, _______, _______, _______, _______, _______,
    _______, _______, _______, _______, _______, _______,                   _______, _______, _______, _______, _______, _______,
    _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
                               _______, _______, _______, _______, _______,  _______, _______, _______
    )
```

File: keyboards/varys/keymaps/via/keymap.c [hunk 17]
```diff
@@ -0,0 +1,76 @@
...
+};
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+    state = update_tri_layer_state(state, _RAISE, _LOWER, _MISC);
+    return state;
+}
+
```
Comment (waffle87) [inline keyboards/varys/keymaps/via/keymap.c:65]:
```suggestion
```

File: keyboards/varys/keymaps/via/keymap.c [hunk 18]
```diff
@@ -0,0 +1,76 @@
...
+#ifdef ENCODER_MAP_ENABLE
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
+    [_QWERTY] = {ENCODER_CCW_CW(KC_BRIGHTNESS_DOWN, KC_BRIGHTNESS_UP),
+                 ENCODER_CCW_CW(KC_VOLD, KC_VOLU)},
+    [_RAISE] = {ENCODER_CCW_CW(_______, _______),
+                ENCODER_CCW_CW(_______, _______)},
+    [_LOWER] = {ENCODER_CCW_CW(_______, _______),
+                ENCODER_CCW_CW(_______, _______)},
+    [_MISC] = {ENCODER_CCW_CW(_______, _______),
+               ENCODER_CCW_CW(_______, _______)}};
```
Comment (waffle87) [inline keyboards/varys/keymaps/via/keymap.c:75]:
```suggestion
                ENCODER_CCW_CW(_______, _______)}
};
```

File: keyboards/varys/readme.md [hunk 19]
```diff
@@ -0,0 +1,32 @@
+# Varys
+
+![varys](https://i.imgur.com/XDd9HKF.jpeg)
+
+Varys is a mechanical split keyboard with custom designed PCB's, embedded ATmega32U4 microcontroller, EC11 encoders, USB-C connectors(for both power and communication), Serial/I2C communication, 132x32 Oled and per-key RGB. Keyboard layout is based on the lily58 layout.
+
+Due to limited firmware space, RGB features have been disabled.
+
+* Keyboard Maintainer: [Vinay Komaravolu](https://github.com/VinayKomaravolu)
+* Hardware Supported: Varys PCB (Atmega324U)
```
Comment (waffle87) [inline keyboards/varys/readme.md:10]:
```suggestion
* Hardware Supported: Varys PCB (Atmega32u4)
```

File: keyboards/varys/varys.c [hunk 20]
```diff
@@ -0,0 +1,64 @@
...
+0xfe, 0xff, 0xc3, 0x99, 0x3c, 0x76, 0xef, 0x9f, 0xbf, 0xff, 0xf8, 0x00, 0x00, 0x00, 0x00, 0x00, 
+0x00, 0x00, 0x00, 0x7f, 0x0f, 0x07, 0xff, 0xff, 0xff, 0x00, 0xff, 0xff, 0x00, 0xff, 0xff, 0x80, 
+0x7f, 0xff, 0xfd, 0xfb, 0x03, 0xfc, 0xff, 0xff, 0xff, 0x0f, 0x7f, 0x7e, 0x00, 0x00, 0x00, 0x00, 
+0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x07, 0x1f, 0x7f, 0xfc, 0xf9, 0xf3, 0xe7, 0xcc, 0x99, 0x73, 
+0x67, 0xee, 0xed, 0xf1, 0xfc, 0xff, 0x3f, 0x0f, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 
+0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x06, 0x06, 0x06, 0x03, 0x07, 0x07, 0x04, 0x00, 0x00, 0x00, 
+0x00, 0x00, 0x06, 0x07, 0x07, 0x06, 0x06, 0x06, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00
+  };
+
+oled_rotation_t oled_init_user(oled_rotation_t rotation) {
```
Comment (waffle87) [inline keyboards/varys/varys.c:30]:
```suggestion
oled_rotation_t oled_init_kb(oled_rotation_t rotation) {
```

File: keyboards/varys/varys.c [hunk 21]
```diff
@@ -0,0 +1,64 @@
...
+    return OLED_ROTATION_270;  // flips the display 180 degrees if offhand
+}
+
+bool oled_task_kb(void) {
```
Comment (waffle87) [inline keyboards/varys/varys.c:34]:
```suggestion
bool oled_task_kb(void) {
    if (!oled_task_user()) {
        return false;
    }
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Comment by vinaykomaravolu** [comment]
@waffle87 I don't understand the suggestion of removing _MISC? I need to it work with my design with the OLED and with VIA. I also need it for how it will be used for the user experience side of things. I need 4 layers.

File: keyboards/varys/keymaps/via/keymap.c (hunk 18 above)
Comment (vinaykomaravolu) [reply keyboards/varys/keymaps/via/keymap.c:75]: need to keep _MISC

File: keyboards/varys/keyboard.json (hunk 5 above)
Comment (vinaykomaravolu) [reply keyboards/varys/keyboard.json:5]: This pcb is not using a promicro as this is a custom pcb with an embedded ATmega32u. I do not think it should be added.

**Comment by waffle87** [comment]
> @waffle87 I thank you for the suggestions but I don't understand removing _MISC? I need to it work with my design with the OLED and with VIA. I also need it for how it will be used for the user experience side of things. I require 4 layers and _MISC for my changes, therefore I would like to leave them in.

Blank layers are discouraged. If you could find a use for it, that would be preferable.

File: keyboards/varys/keyboard.json (hunk 5 above)
Comment (waffle87) [reply keyboards/varys/keyboard.json:6]: And microcontroller is using a caterina bootloader? I find that unlikely. It's more likely to be `atmel-dfu`.

**Comment by waffle87** [comment]
Please do not mark suggestions as resolved without addressing them. Eg. the suggestions regarding using the core tri layer feature...

File: keyboards/varys/keyboard.json (hunk 5 above)
Comment (vinaykomaravolu) [reply keyboards/varys/keyboard.json:6]: Hmmm I did some verification and doubled checked. It is indeed using the caterina bootloader for my ATMEGA32U4-MU MCU. Flashing through QMK MSYS also flashes for bootloader: caterina.
image

**Comment by vinaykomaravolu** [comment]
> > @waffle87 I thank you for the suggestions but I don't understand removing _MISC? I need to it work with my design with the OLED and with VIA. I also need it for how it will be used for the user experience side of things. I require 4 layers and _MISC for my changes, therefore I would like to leave them in.
> 
> Blank layers are discouraged. If you could find a use for it, that would be preferable.

Understood, let me make these changes and resubmit.

**Comment by vinaykomaravolu** [comment]
> Please do not mark suggestions as resolved without addressing them. Eg. the suggestions regarding using the core tri layer feature...

I apologize! I will try to be more aware going forward.

File: keyboards/varys/keymaps/via/keymap.c (hunk 17 above)
Comment (vinaykomaravolu) [reply keyboards/varys/keymaps/via/keymap.c:65]: Correct me if I'm wrong but does the TRI_LAYER_ENABLE = yes feature work with VIA? I've been messing around with this for a while and I can't seem to get it to work for via (It works just fine for default...). layer_state_set_user fixes this issue, lets keep this in for varys/keymaps/via? You can see the recent commit!

**Comment by vinaykomaravolu** [comment]
Addressed all the issues suggested by @waffle87

File: keyboards/varys/keymaps/via/keymap.c (hunk 17 above)
Comment (drashna) [reply keyboards/varys/keymaps/via/keymap.c:62]: VIA enables the feature by default.  In both cases you need to use the specific keycodes. It doesn't work with LT or other keycodes.

Comment (vinaykomaravolu) [reply keyboards/varys/keymaps/via/keymap.c:62]: Understood, let me make the change real quick

Comment (vinaykomaravolu) [reply keyboards/varys/keymaps/via/keymap.c:62]: Let me see what I can do

**Comment by vinaykomaravolu** [comment]
@drashna @waffle87 After your suggestions I went into a deep dive through the code to verify everything. After a couple of hours of debugging, VIA and the OLED would just not work with TRI_LAYER feature without the use of:

layer_state_t layer_state_set_user(layer_state_t state) {
    state = update_tri_layer_state(state, _RAISE, _LOWER, _MISC);
    return state;
}

Not sure if its a bug or a specification with my board. However the most recent commit provides the exact configuration I want the keyboard to be in and works exactly as we intended. 

If these changes are good to go, I'm happy with the merge.

**Comment by vinaykomaravolu** [comment]
@drashna @waffle87  Sorry for the ping. Any updates regarding this PR?

**Pull request closed without merging**